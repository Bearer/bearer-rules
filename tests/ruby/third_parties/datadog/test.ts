const { createInvoker, getEnvironment } = require("../../../helper.js")
const { ruleId, ruleFile, testBase } = getEnvironment(__dirname)

describe(ruleId, () => {
  const invoke = createInvoker(ruleId, ruleFile, testBase)
  
  test.concurrent("datatype_in_tags", async () => {
    const testCase = "datatype_in_tags.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test.concurrent("ok_datatype_ignored", async () => {
    const testCase = "ok_datatype_ignored.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
  test.concurrent("ok_no_datatype", async () => {
    const testCase = "ok_no_datatype.rb"
    expect(invoke(testCase)).toMatchSnapshot();
  })
  
})