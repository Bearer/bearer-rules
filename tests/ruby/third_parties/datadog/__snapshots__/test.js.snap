// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ruby_third_parties_datadog datatype_in_tags 1`] = `
"{
  "high": [
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 3,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 2,
        "end": 2,
        "column": {
          "start": 12,
          "end": 40
        }
      },
      "sink": {
        "start": 3,
        "end": 3,
        "column": {
          "start": 3,
          "end": 16
        },
        "content": "c.tags = user"
      },
      "parent_line_number": 3,
      "snippet": "c.tags = user",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_0",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_0",
      "code_extract": "  c.tags = user"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 7,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 7,
        "end": 7,
        "column": {
          "start": 28,
          "end": 38
        }
      },
      "sink": {
        "start": 7,
        "end": 7,
        "column": {
          "start": 1,
          "end": 39
        },
        "content": "span.set_tag('user.email', user.email)"
      },
      "parent_line_number": 7,
      "snippet": "span.set_tag('user.email', user.email)",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_1",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_1",
      "code_extract": "span.set_tag('user.email', user.email)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 9,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 54,
          "end": 64
        }
      },
      "sink": {
        "start": 9,
        "end": 9,
        "column": {
          "start": 1,
          "end": 65
        },
        "content": "Datadog::Tracing.active_span&.set_tag('customer.id', user.email)"
      },
      "parent_line_number": 9,
      "snippet": "Datadog::Tracing.active_span&.set_tag('customer.id', user.email)",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_2",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_2",
      "code_extract": "Datadog::Tracing.active_span&.set_tag('customer.id', user.email)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 10,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 10,
        "end": 10,
        "column": {
          "start": 53,
          "end": 63
        }
      },
      "sink": {
        "start": 10,
        "end": 10,
        "column": {
          "start": 1,
          "end": 64
        },
        "content": "Datadog::Tracing.active_span.set_tag('customer.id', user.email)"
      },
      "parent_line_number": 10,
      "snippet": "Datadog::Tracing.active_span.set_tag('customer.id', user.email)",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_3",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_3",
      "code_extract": "Datadog::Tracing.active_span.set_tag('customer.id', user.email)"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 12,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 12,
        "end": 12,
        "column": {
          "start": 47,
          "end": 64
        }
      },
      "sink": {
        "start": 12,
        "end": 14,
        "column": {
          "start": 1,
          "end": 4
        },
        "content": "Datadog::Tracing.trace(\\"web.request\\", tags: { email: user.email }) do |span, trace|\\n  call\\nend"
      },
      "parent_line_number": 12,
      "snippet": "Datadog::Tracing.trace(\\"web.request\\", tags: { email: user.email }) do |span, trace|\\n  call\\nend",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_4",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_4",
      "code_extract": "Datadog::Tracing.trace(\\"web.request\\", tags: { email: user.email }) do |span, trace|\\n  call\\nend"
    },
    {
      "cwe_ids": [
        "201"
      ],
      "id": "ruby_third_parties_datadog",
      "title": "Sensitive data sent to Datadog detected.",
      "description": "## Description\\nLeaking sensitive data to third-party loggers is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Datadog.\\n\\n## Remediations\\n\\nWhen logging errors or events, ensure all sensitive data is removed.\\n\\n## Resources\\n- [Datadog docs](https://docs.datadoghq.com)\\n- [Scrubbing data](https://docs.datadoghq.com/tracing/configure_data_security/?tab=mongodb#scrub-sensitive-data-from-your-spans)\\n",
      "documentation_url": "https://docs.bearer.com/reference/rules/ruby_third_parties_datadog",
      "line_number": 17,
      "full_filename": "/tmp/bearer-scan/datatype_in_tags.rb",
      "filename": ".",
      "data_type": {
        "category_uuid": "cef587dd-76db-430b-9e18-7b031e1a193b",
        "name": "Email Address"
      },
      "category_groups": [
        "PII",
        "Personal Data"
      ],
      "source": {
        "start": 17,
        "end": 17,
        "column": {
          "start": 30,
          "end": 40
        }
      },
      "sink": {
        "start": 17,
        "end": 17,
        "column": {
          "start": 3,
          "end": 41
        },
        "content": "span.set_tag('user.email', user.email)"
      },
      "parent_line_number": 17,
      "snippet": "span.set_tag('user.email', user.email)",
      "fingerprint": "2eba6887e2962e1904b9ac5c8be1515a_5",
      "old_fingerprint": "d7f79a4c66507c70a8cf7f2aa6bb1384_5",
      "code_extract": "  span.set_tag('user.email', user.email)"
    }
  ]
}"
`;

exports[`ruby_third_parties_datadog ok_datatype_ignored 1`] = `"{}"`;

exports[`ruby_third_parties_datadog ok_no_datatype 1`] = `"{}"`;
